pymodbus/repl/client/main.py

```diff
@@ -178,7 +178,7 @@ def _process_args(args, string=True):
                             complete_while_typing=True,
                             bottom_toolbar=bottom_toolbar,
                             key_bindings=use_keys,
-                            history=FileHistory('../.pymodhis'),
+                            history=FileHistory('.pymodhis'),
```

Comment:
First of all thanks for the PR !
Your solution however do cause problems on Mac if you install it as an application (as I do) because the bin dir (macOS) is protected against changes.

For that reason I suggest this change, which is inline with @dhoomakethu :
```suggestion
                            history=FileHistory('~/.pymodhis'),
```
If you agree accept the suggestion and I will merge the PR.

Reply:
we will have to make sure  `FileHistory` can resolve `~` properly on windows as well.

Reply:
Using `os.path.normpath(os.path.expanduser("~/.pymodhis")` will make the path resolve correctly to the user directory across both Windows and Linux, if that's the desired behavior.  Sorry, don't have a Mac to test that on.

Reply:
I've made a commit with that added - I've also move the definition of the history file path to the top of `def cli():` block so that it's easier to change or to move to a configuration option or command line flag in the future.